pkg/volume/configmap/configmap.go

```diff
@@ -201,15 +202,20 @@ func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string
 		for _, ktp := range mappings {
 			content, ok := configMap.Data[ktp.Key]
 			if !ok {
-				glog.Errorf("references non-existent config key")
-				return nil, fmt.Errorf("references non-existent config key")
+				err_msg := "references non-existent config key"
+				glog.Errorf(err_msg)
```

Comment:
On Fri, Jul 29, 2016 at 01:40:24PM -0700, Tim Hockin wrote:
> > @@ -201,15 +202,20 @@ func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string
> >  		for _, ktp := range mappings {
> >  			content, ok := configMap.Data[ktp.Key]
> >  			if !ok {
> > -				glog.Errorf("references non-existent config key")
> > -				return nil, fmt.Errorf("references non-existent config key")
> > +				err_msg := "references non-existent config key"
> > +				glog.Errorf(err_msg)
> 
> In general we should not log *and* return the same error.

Yeah, I saw that. But this is the way it was before. And this is (one) of the
cleanups I had in mind. But, IMHO, should go in another PR.